src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,281 @@
+      val pileupClusters: Seq[PileupGroup] = AggregateSvPileup.toClusters(pileupToNeighbors)
+
+      // Aggregate each cluster
+      val aggregatedPileups: Seq[AggregatedBreakpointPileup] = pileupClusters.map(AggregatedBreakpointPileup.apply)
+
+      // Write to output file
+      aggregatedPileups.foreach(writer.write)
+    }
+
+    writer.close()
+
+  }
+
+}
+
+
+object AggregateSvPileup {
+
+  /** Type alias for a collection of pileups */
+  type PileupGroup = Seq[BreakpointPileup]
+
+  /**
+   * Maps each pileup to the set of other pileups within the distance threshold.
+   *
+   * Note: it is recommended to call this method on a group of pileups that share a common `BreakpointContigOrientation`
+   * to reduce unnecessary run time.
+   *
+   * @param pileups Set of pileups to evaluate for pairwise neighbor relationships
+   * @param maxDist Maximum distance (inclusive) for the left and right coordinates to be considered a neighbor
+   * @return Map of pileup to the set of neighbors
+   */
+  private def pileupToNeighbors(pileups: PileupGroup, maxDist: Int): Map[BreakpointPileup, PileupGroup] = {
+    val map: mutable.Map[BreakpointPileup, List[BreakpointPileup]] = mutable.Map.from(pileups.map((_, Nil)))
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.total >= pileup2.total && isCluster(pileup1, pileup2, maxDist)) {
+        map.put(pileup1, pileup2 :: map(pileup1))
+        map.put(pileup2, pileup1 :: map(pileup2))
+      }
+    }
+    map.toMap
+  }
+
+  /** Returns true if the two breakpoints can be clustered, i.e., they are not the same breakpoint (have different
+   * IDs), they have the same left and right chromosomes and strand, and the distance between their left and right
+   * coordinates is below the threshold (inclusive) */
+  private def isCluster(bp1: BreakpointPileup, bp2: BreakpointPileup, maxDist: Int): Boolean = {
+    bp1.id != bp2.id &&
+      BreakpointContigOrientation(bp1) == BreakpointContigOrientation(bp2) &&
+      Math.abs(bp1.left_pos - bp2.left_pos) <= maxDist &&
+      Math.abs(bp1.right_pos - bp2.right_pos) <= maxDist
+  }
+
+  /**
+   * Converts a set of pairwise neighbor relationships to a set of clusters, where each cluster consists of breakpoints
+   * that can be joined into a connected path via pairwise neighbor relationships.
+   * @param pileupToNeighbors Map of pileup to its set of neighbors
+   * @param existingClusters An existing set of clusters to add to
+   * @return The set of pileup clusters
+   */
+  @tailrec
+  def toClusters(pileupToNeighbors: Map[BreakpointPileup, PileupGroup],
+                 existingClusters: Seq[PileupGroup] = Seq()): Seq[PileupGroup] = {
+    if (pileupToNeighbors.isEmpty) {
+      existingClusters
+    } else {
+      val (pileupGroup, remaining) = extractClusterFor(pileupToNeighbors.keys.head, pileupToNeighbors)
+      toClusters(remaining, existingClusters :+ pileupGroup)
+    }
```

Comment:
More idiomatic since you want to avoid using `.head` on a list (like avoiding `.get` on an option)
```suggestion
    pileupToNeighbors.keys.headOption match {
      case None         => existingClusters
      case Some(pileup) =>
        val (pileupGroup, remaining) = extractClusterFor(pileup, pileupToNeighbors)
        toClusters(remaining, existingClusters :+ pileupGroup)
    }
```